pkg/deploy/dev-workspace-config/dev_workspace_config.go

```diff
@@ -146,19 +148,60 @@ func updateWorkspaceStorageConfig(devEnvironments *chev2.CheClusterDevEnvironmen
 	return nil
 }
 
-func updateWorkspaceServiceAccountConfig(devEnvironments *chev2.CheClusterDevEnvironments, workspaceConfig *controllerv1alpha1.WorkspaceConfig) error {
-	isNamespaceAutoProvisioned := pointer.BoolPtrDerefOr(devEnvironments.DefaultNamespace.AutoProvision, constants.DefaultAutoProvision)
+func updateWorkspaceServiceAccountConfig(devEnvironments *chev2.CheClusterDevEnvironments, workspaceConfig *controllerv1alpha1.WorkspaceConfig) {
+	isNamespaceAutoProvisioned := pointer.BoolDeref(devEnvironments.DefaultNamespace.AutoProvision, constants.DefaultAutoProvision)
 
 	workspaceConfig.ServiceAccount = &controllerv1alpha1.ServiceAccountConfig{
 		ServiceAccountName:   devEnvironments.ServiceAccount,
 		ServiceAccountTokens: devEnvironments.ServiceAccountTokens,
 		// If user's Namespace is not auto provisioned (is pre-created by admin), then ServiceAccount must be pre-created as well
-		DisableCreation: pointer.BoolPtr(!isNamespaceAutoProvisioned && devEnvironments.ServiceAccount != ""),
+		DisableCreation: pointer.Bool(!isNamespaceAutoProvisioned && devEnvironments.ServiceAccount != ""),
 	}
-	return nil
 }
 
 func updateWorkspacePodSchedulerNameConfig(devEnvironments *chev2.CheClusterDevEnvironments, workspaceConfig *controllerv1alpha1.WorkspaceConfig) error {
 	workspaceConfig.SchedulerName = devEnvironments.PodSchedulerName
 	return nil
 }
+
+func updateProjectCloneConfig(devEnvironments *chev2.CheClusterDevEnvironments, workspaceConfig *controllerv1alpha1.WorkspaceConfig) {
+	if devEnvironments.ProjectCloneContainer == nil {
+		return
+	}
+	if workspaceConfig.ProjectCloneConfig == nil {
+		workspaceConfig.ProjectCloneConfig = &controllerv1alpha1.ProjectCloneConfig{}
+	}
+	container := devEnvironments.ProjectCloneContainer
+
+	workspaceConfig.ProjectCloneConfig.Image = container.Image
+	workspaceConfig.ProjectCloneConfig.ImagePullPolicy = container.ImagePullPolicy
+	workspaceConfig.ProjectCloneConfig.Env = container.Env
+	workspaceConfig.ProjectCloneConfig.Resources = cheResourcesToCoreV1Resources(container.Resources)
+}
+
+func cheResourcesToCoreV1Resources(resources *chev2.ResourceRequirements) *corev1.ResourceRequirements {
```

Comment:
I have some concerns:
* No value (`nil`) in `resources` should lead to setting defaults
* Zero value (`"0"`) should not lead to setting any limits/requests

Reply:
I believe that's the current behavior here, though maybe indirectly. Since DWO controls the project-clone container, it also holds the default values and handling of zero values -- implementing this logic here would work against us:
* Che Operator does not know defaults for project-clone, and shouldn't set them
* If we convert `"0"` into "do not set field here", it will lead to the DWOC not having the field set, which will in turn lead to DWO using the default value

Basically, DWO and Che implement similar logic for handling resources overrides, and so on the Che Operator side, we need to just pass all fields from the CheCluster into the DevWorkspaceOperatorConfig, unchanged (which requires copying fields in Go since the structs come from different packages).

I think maybe it would be clearer if the function had a better name -- this is intended mostly for converting from the `chev2` struct to the `corev1` struct directly to allow passing it (unmodified) to the DevWorkspace Operator.

Reply:
> I believe that's the current behavior here, though maybe indirectly. Since DWO controls the project-clone container, it also holds the default values and handling of zero value

Yes, I found it [1], not sure about settings defaults. Any way, I think it make sense to put some comments here:
`Since DWO controls the project-clone container, it also holds the default values and handling of zero value`

[1] https://github.com/devfile/devworkspace-operator/blob/68fbeb46519c59b6786e4716382913c6fea8272a/pkg/library/projects/clone.go#L97

Reply:
Good idea -- applied